### DISCUSSION TITLE
How to query json column with supabase client?

### DISCUSSION THREAD

**Discussion by mstade** [supabase/supabase#599] [Questions] [answered]
Hi!

I have a simple query on table where a column imaginatively name `data` is of type `jsonb`:

```sql
select data from userinfo where data ->> 'personalNumber' = '12345'
```

This works fine in the query editor in supabase. However, when using the JS client I can't seem to figure out how the equivalent query. Running this yields an error:

```javascript
var { data, error } = await supabase
	.from('userinfo')
	.select('data')
	.eq(`data ->> 'personalNumber'`, person.user.personalNumber)
```

The error is:

```javascript
{
  hint: 'Perhaps you meant to reference the column "userinfo.data".',
  details: null,
  code: '42703',
  message: 'column userinfo.data  does not exist'
}
```

How would you write this query using the supabase client?

**Comment by inian** [comment, marked as answer]
Can you try with `data->personalNumber`? Here are the docs for querying JSON with the JS client - https://supabase.io/docs/client/select#quering-json-data

**Reply by mstade** [reply]
That worked – thank you @inian! So it turns out you mustn't use spaces or quotes when querying JSON, which was a bit of a gotcha given I used both in the example query. If you use spaces you get the "column doesn't exist" error and if you use quotes after removing the spaces you get a syntax error. Might be worth adding to the docs as a pro-tip.

Do you accept pull requests on the docs?
